integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -235,14 +237,38 @@ private[sql] case class CarbonProjectForUpdateCommand(
         }
         sys.error("Update operation failed. please check logs.")
     } finally {
-      if (null != dataSet && isPersistEnabled) {
-        dataSet.unpersist()
+      if (updateLock.unlock()) {
+        LOGGER.info(s"updateLock unlocked successfully after update operation $tableName")
+      } else {
+        LOGGER.error(s"Unable to unlock updateLock for table $tableName after table updation");
```

Comment:
```suggestion
        LOGGER.error(s"Unable to unlock updateLock for table $tableName after table update");
```

Reply:
I have modified code according to your suggestion